corehq/apps/events/es.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from django.core.paginator import Paginator
+
+from corehq.apps.es import CaseES, CaseSearchES
+from corehq.apps.es.case_search import case_property_text_query
+from corehq.apps.events.models import (
+    LOCATION_IDS_CASE_PROPERTY,
+    get_attendee_case_type,
+)
+from corehq.form_processor.models import CommCareCase
+
+
+class AttendeeSearchES(CaseSearchES):
+    """
+    Allows Attendee cases to be searched by location, like Users.
+    """
+    @property
+    def builtin_filters(self):
+        return [
+            attendee_location_filter,
+        ] + super().builtin_filters
+
+
+def attendee_location_filter(location_ids):
+    """
+    Return Attendee cases whose location IDs include at least one of
+    ``location_ids``.
+    """
+    if not isinstance(location_ids, str):
+        location_ids = ' '.join(location_ids)
+    return case_property_text_query(
+        LOCATION_IDS_CASE_PROPERTY,
+        location_ids,
+        operator='or',  # Any location ID will be a hit
+    )
+
+
+def get_paginated_attendees(domain, limit, page, query=None):
+    assert page > 0, 'Page numbering starts at 1'
```

Comment:
How does an AssertionError translate to the FE?

Reply:
I presume it shouldn't really and is meant to be picked up during dev?

Reply:
The frontend never sees these assertion errors because the view always calls this function with valid values (`page` defaults to 1 and `limit` defaults to 10).

Reply:
"never" :thinking: A user _can_ find how this URL is called, and then manually set the GET params, if they wanted to. And then they would get a 500.

Reply:
Coolzies, thanks. I was curious